internal/controller/dnsrecord_controller.go

```diff
@@ -387,19 +401,16 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 				return noRequeueDuration, err
 			}
 		}
-		dnsRecord.Status.ValidFor = validationRequeueTime.String()
 		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, "AwaitingValidation", "Awaiting validation")
-		logger.Info("Applying changes")
-		err = registry.ApplyChanges(ctx, plan.Changes)
-		if err != nil {
-			return validationRequeueTime, err
-		}
 	} else {
 		logger.Info("All records are already up to date")
 		dnsRecord.Status.WriteCounter = 0
 		wrtiteCounter.WithLabelValues(dnsRecord.Name, dnsRecord.Namespace).Set(0)
 		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionTrue, "ProviderSuccess", "Provider ensured the dns record")
 	}
 
+	dnsRecord.Status.ValidFor = defaultRequeueTime.String()
+	dnsRecord.Status.QueuedAt = reconcileStart
+
```

Comment:
I think I found a bug in the way it was before :laughing:  Ok, so we are never really returning the `validationRequeueTime` from the `aaplyChanges` function. The reason it was passing tests - integration suite was able to get validation on the `defaultValidation`. Basically the `.Status.ValidFor` is a return value for the validation time. In the changes you are proposing we are setting that status always to the `defaultRequeuTime` while we should set it to either `ValidationRequeueTime` or `defaultRequeueTime` based on if we had changes or not and we should use the appropriate return value.